samples/openthread/coap_server/CMakeLists.txt

```diff
@@ -10,6 +10,8 @@ set(NRF_SUPPORTED_BOARDS
 	nrf52833dk_nrf52833
 )
 
+set(CONF_FILE "prj.conf ../openthread.conf")
```

Comment:
This is not acceptable, users should always be able to specify `-DCONF_FILE=<somefile.conf>` on command line when invoking CMake (or `west build`)

Reply:
It's always possible to pass more than one file in `CONF_FILE`: the https://docs.zephyrproject.org/latest/guides/kconfig/setting.html?highlight=-DCONF_FILE#the-initial-configuration

Reply:
not in this case, cause you overwrite whatever setting user provides.
If user does:
```
-DCONF_FILE=custom.conf
```
then that setting is overwritten by this line:
```
set(CONF_FILE "prj.conf ../openthread.conf")
```

Try:
```
+message("CONF_FILE ${CONF_FILE}")
 set(CONF_FILE "prj.conf ../openthread.conf")
+message("CONF_FILE ${CONF_FILE}")
```
and see for yourself.

Reply:
Would this help?

`set(CONF_FILE CACHE STRING "prj.conf ../openthread.conf")`

Reply:
no, that would put it directly in the cache, and thus break the environment `CONF_FILE` support.
also remember, user may do `-DOVERLAY_CONFIG=<file>` which may also end up colliding with the proposal when you set multiple files.